### ISSUE TITLE
JsonProperty required allowed but not respected on record parameters

### ISSUE THREAD

**Issue by mikethea1** [JamesNK/Newtonsoft.Json#2951]
### Source/destination types

```csharp
record R([JsonProperty(Required = Required.Always)] string A, [JsonProperty(Required = Required.Always)] string[] B);
```

### Source/destination JSON

```javascript
{}
```

### Expected behavior

Deserialization error because the required properties were not present.

### Actual behavior

Deserializes successfully with all properties null.

### Steps to reproduce

```csharp
JsonConvert.DeserializeObject<R>("{}");
```

**Comment by MO2k4** [comment]
Your attribute is invalid, try to use [property:JsonProperty...]

**Comment by elgonzo** [comment]
@MO2k4 

No, the application of the attribute as shown is **not** invalid. It's meant to support constructor parameters:

https://github.com/JamesNK/Newtonsoft.Json/blob/2eaa475f8853f2b01ac5591421dcabd7a44f79ce/Src/Newtonsoft.Json/JsonPropertyAttribute.cs#L34-L35
